core/src/server/rpc_module.rs

```diff
@@ -775,101 +779,73 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 	}
 }
 
-/// Represent a pending subscription which waits to be accepted or rejected.
-///
-/// Note: you need to call either `PendingSubscription::accept` or `PendingSubscription::reject` otherwise
-/// the subscription will be dropped with an `InvalidParams` error.
+/// Represents a single subscription.
 #[derive(Debug)]
-struct InnerPendingSubscription {
+pub struct SubscriptionSink {
 	/// Sink.
-	sink: MethodSink,
+	inner: MethodSink,
 	/// Get notified when subscribers leave so we can exit
 	close_notify: Option<SubscriptionPermit>,
 	/// MethodCallback.
 	method: &'static str,
+	/// Shared Mutex of subscriptions for this method.
+	subscribers: Subscribers,
 	/// Unique subscription.
 	uniq_sub: SubscriptionKey,
-	/// Shared Mutex of subscriptions
-	subscribers: Subscribers,
-	/// Request ID.
-	id: Id<'static>,
+	/// Id of the subscription.
+	///
+	/// *Note*: Having some value means the subscription was not accepted or rejected yet.
+	id: Option<Id<'static>>,
+	/// Returns when the unsubscribe method has been called.
+	///
+	/// *Note*: Have some values means the subscription was accepted.
+	unsubscribe: Option<watch::Receiver<()>>,
```

Comment:
add a type alias for that I reckon that should make the code more readable and the comment should not be necessary